### PR TITLE
Changed reference style in dbg macro docs.

### PR DESCRIPTION
**Pull request by DevQps**
# Description

A continuation of Pull Request #59528 :
- Fixed method of referencing and adjusted the references as suggested by @lzutao

### CODE REVIEW

**Comment by DevQps** [comment]
@lzutao @Centril This is the follow-up PR!

**Comment by Centril** [comment]
@pietroalbini highfive is being unreasonable...

r? @Centril

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -315,8 +315,8 @@ macro_rules! eprintln {
 /// file and line whenever it's reached.
 ///
 /// [stderr]: https://en.wikipedia.org/wiki/Standard_streams#Standard_error_(stderr)
-/// [debug-log]: https://docs.rs/log/*/log/macro.debug.html
-/// [log]: https://docs.rs/log/
+/// [`debug`]: https://docs.rs/log/*/log/macro.debug.html
```
Comment (lzutao) [inline src/libstd/macros.rs]: Missing exclamation `!`.

Comment (DevQps) [reply src/libstd/macros.rs]: Pffff..... I feel annoyed with myself that I forget such simple things...

Comment (lzutao) [reply src/libstd/macros.rs]: No worry! You're doing really well.

Comment (DevQps) [reply src/libstd/macros.rs]: Thanks! Truly trying my best here :)

**Pull request merged**